src/main/java/com/cloudant/client/api/DesignDocumentManager.java

```diff
@@ -0,0 +1,213 @@
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+/**
+ * Provides API to work with design documents.
+ * <h3>Usage Example:</h3>
+ * <pre>
+ * // read from javascript design document file
+ * {@code
+ * DesignDocument design = DesignDocumentManager.fromFile("../example.js");
+ * }
+ *
+ * // read from directory of design documents
+ * {@code
+ * List<DesignDocument> design =
+ *                  DesignDocumentManager.fromDirectory("../design-files/");
+ * }
+ *
+ * // sync with the database
+ * {@code
+ * db.getDesignDocumentManager().put(designDoc);
+ * }
+ *
+ * // sync multiple design documents with the database
+ * {@code
+ * db.getDesignDocumentManager().put(listDesignDocuments);
+ * }
+ *
+ * // read from the database
+ * {@code
+ * DesignDocument designDoc2 = db.getDesignDocumentManager().get("_design/example");
+ * }
+ * </pre>
+ *
+ * @see {@link Database#getDesignDocumentManager()} to access the API.
+ * @see DesignDocument
+ * @since 2.0.0
+ */
+public class DesignDocumentManager {
+
+    private URI dbUri;
+    private Database db;
+
+    DesignDocumentManager(Database db) {
+        this.dbUri = db.getDBUri();
+        this.db = db;
+    }
+
+    /**
+     * Synchronizes a design document to the Database.
+     * <p>This method will first try to find a document in the database with the same id
+     * as the given document, if it is not found then the given document will be saved to the
+     * database.
+     * <p>If the document was found in the database, it will be compared with the given document
+     * using {@code equals()}. If both documents are not equal, then the given document will be
+     * saved to the database and updates the existing document.
+     *
+     * @param document The design document to synchronize
+     * @return {@link Response} as a result of a document save or update, or returns {@code null}
+     * if no action was taken and the document in the database is up-to-date with the given document.
+     */
+    public Response put(DesignDocument document) {
+        CouchDbUtil.assertNotEmpty(document, "Document");
+        DesignDocument documentFromDb;
+        try {
+            documentFromDb = get(document.getId());
+
+        } catch (NoDocumentException e) {
+            return db.save(document);
+        }
+        if (!document.equals(documentFromDb)) {
+            document.setRevision(documentFromDb.getRevision());
+            return db.update(document);
+        }
+        return null;
+    }
+
+    public void put(DesignDocument... designDocs) {
+        for (DesignDocument designDocument : designDocs) {
+            put(designDocument);
+        }
+    }
+
+    /**
+     * Gets a design document from the database.
+     *
+     * @param id The document id
+     * @return {@link DesignDocument}
+     */
+    public DesignDocument get(String id) {

```

Comment:
For places that take an `id`, how do you feel about prefixing `_design` if it is absent?
I think that is the approach we took for view references to design docs.